base/abstractset.jl

```diff
@@ -491,7 +491,8 @@ _overlapping_range_isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T
 
 function _overlapping_range_isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Integer
     if abs(step(a)) == abs(step(b))
-        return mod(minimum(a) - minimum(b), step(a)) != 0
+        m1, m2 = minmax(minimum.((a,b))...)
+        return mod(m2 - m1, step(a)) != 0
```

Comment:
This can still overflow for large `m2` and large negative `m1` (e.g. `isdisjoint(typemax(Int8):Int8(3):typemax(Int8), typemin(Int8):Int8(3):typemax(Int8))`). For `BitInteger`s, `unsigned(m2 - m1)` would work, but not all Integers support `unsigned`. `unsigned(::BigInt)`, for example, throws.

I think that computing mod twice instead of subtracting would work, though it is marginally slower

```suggestion
        return mod(minimum(a), step(a)) != mod(minimum(b), step(a))
```